packages/@aws-cdk/aws-lambda-event-sources/lib/stream.ts

```diff
@@ -21,6 +21,47 @@ export interface StreamEventSourceProps {
    */
   readonly batchSize?: number;
 
+  /**
+   * If the function returns an error, split the batch in two and retry.
+   *
+   * @default false
+   */
+  readonly bisectBatchOnFunctionError?: boolean;
+
+  /**
+   * An Amazon SQS queue or Amazon SNS topic destination for discarded records.
+   *
+   * @default discarded records are ignored
+   */
+  readonly destinationConfig?: lambda.CfnEventSourceMapping.DestinationConfigProperty | IResolvable | undefined;
```

Comment:
Couple of points here -

1. We shouldn't be using union types in the CDK. CDK uses JSII to generate language bindings in other languages such as Java, C# and Python, most of whom don't have any form of union types. This ends up being `java.lang.Object` (in the case of Java) which generates a poor customer experience. 

2. Hand managed CDK libraries (such as this one) should not be expose any of the autogenerated `Cfn*` classes or its underlying types. We might have to build classes similar to this - https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-lambda-destinations/lib - or perhaps reuse them. I suggest into looking at what's more feasible/cleaner here.

Reply:
I ended up re-using the classes from lambda destinations and following that pattern.  If you don't mind could you give that a quick sanity check - if it looks good I'm planning on getting some logic in there to make sure that the destination was an SQS or SNS queue as those are the only allowed types AFAIK.